api/src/npm/specifiers.rs

```diff
@@ -68,18 +69,18 @@ impl<'a> SpecifierRewriter<'a> {
       }
     }
 
-    if *resolved_specifier == *specifier {
-      // No need to rewrite if the specifier is the same as the resolved
-      // specifier.
-      return None;
-    }
-
     let new_specifier = if resolved_specifier.scheme() == "file" {
       relative_import_specifier(self.base_specifier, &resolved_specifier)
     } else {
       resolved_specifier.to_string()
     };
 
+    if &new_specifier == source_text_specifier {
```

Comment:
This was happening in the wrong place. If we need to update the specifier because the base specifier is different, we still need to write the updated specifier.